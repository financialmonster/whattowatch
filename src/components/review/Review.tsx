import React, { FC, memo } from 'react';
import { Map } from 'immutable';    

import { mapMonthToNubmer } from 'utils';

type TReviewProps = {
    review: Map<string, any>;
}

export const Review: FC<TReviewProps> = memo(({ review }) => {
    const date = new Date(Date.parse(review.get(`date`)));
    const year = date.getFullYear();
    const month = date.getMonth();
    const day = date.getDate();

    return (
        <div className="review">
            <blockquote className="review__quote">
                <p className="review__text">
                    {review.get(`comment`)}
                </p>
                <footer className="review__details">
                    <cite className="review__author">
                        {review.getIn([`user`, `name`])}
                    </cite>
                    <time className="review__date" dateTime={`${year}-${month + 1}-${day}`}>
                        {`${mapMonthToNubmer(month)} ${day}, ${year}`}
                    </time>
                </footer>
            </blockquote>
            <div className="review__rating">
                {review.get(`rating`)}
            </div>
        </div>
    );
});
