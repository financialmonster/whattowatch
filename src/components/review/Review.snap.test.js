import React from 'react';
import renderer from 'react-test-renderer';
import { Map } from 'immutable';

import { Review } from './Review';

it(`Review: correctly rendered`, () => {
    const tree = renderer.create(
        <Review review={Map()} />  
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
