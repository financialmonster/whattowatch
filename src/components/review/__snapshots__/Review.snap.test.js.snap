// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review: correctly rendered 1`] = `
<div
  className="review"
>
  <blockquote
    className="review__quote"
  >
    <p
      className="review__text"
    />
    <footer
      className="review__details"
    >
      <cite
        className="review__author"
      />
      <time
        className="review__date"
        dateTime="NaN-NaN-NaN"
      >
        undefined NaN, NaN
      </time>
    </footer>
  </blockquote>
  <div
    className="review__rating"
  />
</div>
`;
